crates/kubelet/src/kubelet.rs

```diff
@@ -162,6 +141,107 @@ impl<P> Clone for Kubelet<P> {
     }
 }
 
+/// Listens for updates to pods on this node and forwards them to queue.
+async fn start_pod_informer<P: 'static + Provider + Sync + Send>(
+    client: kube::Client,
+    node_name: String,
+    mut queue: PodQueue<P>,
+) -> anyhow::Result<()> {
+    let node_selector = format!("spec.nodeName={}", node_name);
+    let params = ListParams {
+        field_selector: Some(node_selector),
+        ..Default::default()
+    };
+    let api = Api::<KubePod>::all(client);
+    let informer = Informer::new(api).params(params);
+    loop {
+        // TODO Should these results be ignored if Err? It seems reasonable to just keep trying,
```

Comment:
I agree here. Just log the error and let it keep trying so this can handle the graceful shutdown

Reply:
Will do.